core/src/main/java/org/jboss/pnc/build/finder/pnc/client/PncUtils.java

```diff
@@ -208,7 +209,7 @@ public static KojiBuild pncBuildToKojiBuild(PncBuild pncBuild) {
         try {
             buildInfo.setId(Integer.parseInt(build.getId()));
         } catch (NumberFormatException e) {
-            buildInfo.setId(-1);
+            buildInfo.setId(Objects.hash(build.getId()));
```

Comment:
I think that this is a partial fix, because I didn't see any build with id -1, but it should show up now.